presto-raptor/src/main/java/com/facebook/presto/raptor/storage/organization/ShardOrganizerUtil.java

```diff
@@ -72,4 +209,68 @@ static long determineDay(long rangeStart, long rangeEnd, Type type)
         long millisInEndDay = rangeEnd - Duration.ofDays(endDay).toMillis();
         return (millisInStartDay >= millisInEndDay) ? startDay : endDay;
     }
+
+    private static Optional<ShardRange> getShardRange(List<TableColumn> columns, ResultSet resultSet)
+            throws SQLException
+    {
+        ImmutableList.Builder<Object> minValuesBuilder = ImmutableList.builder();
+        ImmutableList.Builder<Object> maxValuesBuilder = ImmutableList.builder();
+
+        for (TableColumn tableColumn : columns) {
+            long columnId = tableColumn.getColumnId();
+            Type type = tableColumn.getDataType();
+
+            Object min = getValue(resultSet, type, minColumn(columnId));
+            Object max = getValue(resultSet, type, maxColumn(columnId));
+
+            minValuesBuilder.add(min);
+            maxValuesBuilder.add(max);
+        }
+
+        List<Type> types = columns.stream()
```

Comment:
I think it'd be more clear for this to be inside the loop, so it's easy to see that the min/max/type lists are matched